lib/core/MantraModel.js

```diff
@@ -0,0 +1,97 @@
+import get from 'lodash/get';
+import has from 'lodash/has';
+import set from 'lodash/set';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { error } from './../utils';
+
+export default class MantraModel {
+    constructor({ name, schema }) {
+        this._isConstructorValid({ name, schema });
+        this._path = 'model';
+        const CONFIG = this._getConfig({ name, schema });
+        this.setName(CONFIG);
+        this.setFields(CONFIG, schema);
+    }
+
+    /**
+     * Set the name property
+     * @param {Object} - MantraModel configuration Object
+     */
+    setName(config) {
+        this.name = this.getName(config);
+    }
+
+    /**
+     * Resolve name property
+     * @param {Object} - MantraModel configuration Object
+     * @returns {string || undefined} - Value to be set on the name property.
+     */
+    getName(config) {
+        if (config.name) return config.name;
+        return;
+    }
+
+    /**
+     * Resolve the MantraModelConfig needed from the schema
+     * @param {Object} { name, schema } - The constructor params
+     * @returns {Object} - A MantraModel configuration object.
+     */
+    setFields(config, schema) {
+        const { fields: fieldSource } = schema;
+        const { fields } = config;
+
+        this.fields = fields.reduce((acc, field) => {
+            if (!has(fieldSource, field)){
+                throw new Error( error(`Field ${field} not found in ${schema.name} schema`) );
+            }
+            set(acc, field, cloneDeep(fieldSource[field]));
+            return acc;
+        }, {});
+    };
+
+    /**
+     * Resolve the MantraModelConfig needed from the schema
+     * @param {Object} { name, schema } - The constructor params
```

Comment:
This is a comment rather than the proposition of an immediate change.

Many of Mantra class' methods receive objects as a parameter. JSDoc allows us to describe and detail the parameters that a function can receive. But there's a problem, in case that a parameter is an object we would like to detail the properties it has (because that would make our documentation clearer).

The solution the community has used is to create `types` which allows you to create an "interface" definition for objects. These `types` allows you to define "objects" and detail their properties as well as their data types. This approach is used by Vue.js, Axios, Vee-validate, Jest and many others.

I think that in the future, we should consider using `types` in order to correctly detail the properties of object that's received as a parameter.

With time and as Mantra evolves, we'll see that these parameters description might look like:
``` javascript
@param {Object} { name, schema, prop3, prop4, prop5, ... } - The constructor params
```

But with types it may look like:
```javascript
interface  GetConfigObject {
    name: string,
    schema: object
}
@param {GetConfigObject} payload - The constructor params
```

Note: All of the above are pure examples that I just thought of (not real implementation).